meta-lxatac-bsp/recipes-core/rauc/files/0001-src-nbd-initialize-info_headers-to-NULL.patch

```diff
@@ -0,0 +1,36 @@
+From 967d32571e226d15c345b2d8cec932a651716f4b Mon Sep 17 00:00:00 2001
+From: Enrico Jorns <ejo@pengutronix.de>
+Date: Thu, 18 Jan 2024 16:49:46 +0100
+Subject: [PATCH] src/nbd: initialize info_headers to NULL
+
+When g_variant_dict_lookup() does not find the given key, it does not
+modify the values of the arguments provided.
+
+This results in info_headers being undefined and can cause the
+rauc-nbd-server to crash, which will result in a broken request on
+client side that lets RAUC terminate hard with:
+
+(rauc:769): rauc-nbd-ERROR **: 15:40:30.553: failed to recv nbd config reply header
+
+The behavior of g_variant_dict_lookup() has also recently been clarified
+in the glib documentation:
+https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3334
```

Comment:
You might want to add an [Upstream-Status](https://docs.yoctoproject.org/contributor-guide/recipe-style-guide.html#patch-upstream-status) like:

    Upstream-Status: Submitted [https://github.com/rauc/rauc/pull/1332]